python/ray/data/aggregate.py

```diff
@@ -103,7 +115,62 @@ def _validate(self, schema: Optional[Union[type, "pa.lib.Schema"]]) -> None:
         pass
 
 
-class _AggregateOnKeyBase(AggregateFn):
+@PublicAPI(stability="alpha")
+class AggregateFnV2(AggregateFn):
+    """Provides an interface to implement efficient aggregations to be applied
+    to the dataset.
+
+    `AggregateFnV2` instances are passed to a Dataset's ``.aggregate(...)`` method to
+    perform aggregations by applying distributed aggregation algorithm:
+
+        - `aggregate_block` is applied to individual blocks, producing partial
+            aggregations.
+        - `merge` combines multiple partially aggregated values (previously returned
+            from `aggregate_block` partial aggregations into a singular partial
+            aggregation.
+        - `finalize` transforms partial aggregation into its final state (for
+            some aggregations this is an identity transformation, ie no-op)
```

Comment:
Nit, docstrings for the arguments should go with the __init__ method. So that IDEs can correctly prompt them.